### PR TITLE
Refactor: change rollout's json tag

### LINKED ISSUE #2289: rollout in application status should be ommited if no value
image

Now the status.rollout will always exist in Application status, I hope it could be ommitted if no value. Just change the belowing line to be a struct pointer is fine.

https://github.com/oam-dev/kubevela/blob/324bbe5e57ab95057cf1141170cb80cba6d80140/apis/core.oam.dev/common/types.go#L269

### PR DESCRIPTION
**Pull request by GingoBang**
[Refactor]: set rollout as a pointer reference

set rollout as a type of pointer references so that the output of json decoding will be "nil" when all subfields has a nil or empty value 

Closes #2289

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2289

**Special notes for your reviewer**:

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]
LGTM, please fix CI.

you can run `make reviewable` to see

**Comment by GingoBang** [comment]
> LGTM, please fix CI.
> 
> you can run `make reviewable` to see

I am using kustomize v4.3, the command of "make reviewable"  will delete a lot of yaml files.. should I downgrade it to 3.8.x to avoid these changes?

**Comment by wonderflow** [comment]
@GingoBang Hi, please fix the CI and the conflicts

**Comment by GingoBang** [comment]
> @GingoBang Hi, please fix the CI and the conflicts

yes sure, I am working on it ^^

**Comment by wonderflow** [comment]
@GingoBang Some rollout e2e tests are still failing for this change

**Comment by GingoBang** [comment]
> @GingoBang Some rollout e2e tests are still failing for this change

there's something wrong of my local environment, I am

File: charts/vela-core/templates/addons-default/terraform.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        type: raw
+    status: {}
+  dependsOn:
+  - ref:
+      apiVersion: core.oam.dev/v1beta1
+      kind: Initializer
+      name: fluxcd
+      namespace: vela-system
+status:
+  observedGeneration: 0
```
Comment (wonderflow) [inline charts/vela-core/templates/addons-default/terraform.yaml:36]: don't add this file in this PR, not related

**Review by wonderflow — CHANGES_REQUESTED** [review]

File: charts/vela-core/templates/addons-default/terraform.yaml (hunk 1 above)
Comment (GingoBang) [reply charts/vela-core/templates/addons-default/terraform.yaml]: done

**Review by wonderflow — APPROVED** [review]
good job, thanks!

**Pull request merged**